File: src/shared/golang/backend/machine_state/machine_state.go
package machine_state

import (
	"github.com/jmoiron/sqlx"
)

type MachineStateManager struct {
	db *sqlx.DB
}

func CreateMachineStateManager(db *sqlx.DB) *MachineStateManager {
	return &MachineStateManager{
		db: db,
	}
}

type MachineState struct {
	Id              int64   `db:"id"`
	EngagementId    int64   `db:"engagement_id"`
	UniqueId        string  `db:"unique_id"`
	Hypervisor      *string `db:"hypervisor"`
	OperatingSystem *string `db:"operating_system"`
}

File: src/webapp/client/src/ts/types/machineState.ts
export enum MachineStateType {
    None,
    Hypervisor,
    OperatingSystem
}

export interface MachineState {
    Id              : number
    EngagementId    : number
    UniqueId        : string
    Hypervisor      : string | null
    OperatingSystem : string | null
}

export function createEmptyMachineState() : MachineState {
    return {
        Id: -1,
        EngagementId: -1,
        UniqueId: '',
        Hypervisor: null,
        OperatingSystem: null
    }
}
